lib/msf/ui/console/command_dispatcher/modules.rb

```diff
@@ -95,14 +97,16 @@ def cmd_edit(*args)
               system(*editor.split, path)
 
               # XXX: This will try to reload *any* .rb and break on modules
-              if args.length > 0 && path.end_with?('.rb')
-                print_status("Reloading #{path}")
-                load path
-              else
-                print_error('Only Ruby files can be reloaded (use reload/rerun for modules)')
+              if !editing_module
+                if path.end_with?('.rb')
+                  print_status("Reloading #{path}")
+                  load path
```

Comment:
How does this play if your edit removed a namespace previously defined? Might want to unload the existing one first.

Reply:
No worries, its a development use case anyway for messing with libs in the repl. Thanks for cleaning this up.